sonic-chassisd/scripts/chassisd

```diff
@@ -357,8 +357,8 @@ class ModuleUpdater(logger.Logger):
 
         # In line card push the hostname of the module and num_asics to the chassis state db.
         # The hostname is used as key to access chassis app db entries 
-        module_info_dict = self._get_module_info(my_index)
         if not self._is_supervisor():
+           module_info_dict = self._get_module_info(my_index)
            hostname_key = "{}{}".format(ModuleBase.MODULE_TYPE_LINE, int(self.my_slot) - 1)
```

Comment:
What is the expectation here? This makes an assumption that line card slots must be numbers, but should that be a requirement? Should this be using `my_index` instead? Or should there be some logic to handle if the LC slot was non-numeric?

Reply:
Move the self._get_module_info() call inside if condition since my_slot is only use inside.  Not need to call self._get_module_info(0 if it not required.

Reply:
But this is still requiring that line cards must be named as numbers, only supervisors are allowed to be alphanumeric. Is that expected and required? Why can't alphanumeric line card names also be supported since this was already found to be a limitation of supervisor naming?

Reply:
Yes. Linecard slot must be number string or number.  It is required.  Otherwise, we will not able construct the linecard name.  BTW, this change just moved the _get_module_info() call into "if" section since we only use it inside the "if".

Reply:
Agree. For the long term solution, we should introduce label attribute to support alphanumeric slot.